### PR TITLE
Quota(fix): remove service accounts from quota count

### PR DESCRIPTION
**Pull request by eleijonmarck**
**What this PR does / why we need it**:
We do not count the service account from the UserQuery for Quotas

**Which issue(s) this PR fixes**:
https://github.com/grafana/grafana/issues/56699

Fixes #

**Special notes for your reviewer**:

Added a new issue for potential refactoring into enum - https://github.com/grafana/grafana/issues/56776

### CODE REVIEW

File: pkg/services/sqlstore/quota.go [hunk 1]
```diff
@@ -155,7 +155,7 @@ func (ss *SQLStore) GetUserQuotaByTarget(ctx context.Context, query *models.GetU
 		var used int64
 		if query.Target != alertRuleTarget || query.UnifiedAlertingEnabled {
```
Comment (eleijonmarck) [inline pkg/services/sqlstore/quota.go:156]: wondering if we should also change this to query.Target to ensure we are looking at the users table. Someone could target some other table for `user_ids`

Comment (eleijonmarck) [reply pkg/services/sqlstore/quota.go:156]: @papagian could you comment on this query.Target, do you know what the query.Target is targeting other than user table?

File: pkg/services/sqlstore/quota.go [hunk 2]
```diff
@@ -155,7 +155,7 @@ func (ss *SQLStore) GetUserQuotaByTarget(ctx context.Context, query *models.GetU
...
 			// get quota used.
-			rawSQL := fmt.Sprintf("SELECT COUNT(*) as count from %s where user_id=?", dialect.Quote(query.Target))
+			rawSQL := fmt.Sprintf("SELECT COUNT(*) as count from %s where user_id=? and is_service_account = %s", dialect.Quote(query.Target), ss.Dialect.BooleanStr(false))
```
Comment (kalleep) [inline pkg/services/sqlstore/quota.go:158]: This function is used for more than checking quota for users. Seems like dashboard, alert_rule etc is using this as well. So I am pretty sure we cannot add is_service_account here

File: pkg/services/sqlstore/quota.go (hunk 1 above)
Comment (eleijonmarck) [reply pkg/services/sqlstore/quota.go:166]: this is solved by adding a notServiceAccount function to each target.Query that targets the "user" table

File: pkg/services/sqlstore/quota.go [hunk 3]
```diff
@@ -156,6 +166,10 @@ func (ss *SQLStore) GetUserQuotaByTarget(ctx context.Context, query *models.GetU
 		if query.Target != alertRuleTarget || query.UnifiedAlertingEnabled {
 			// get quota used.
 			rawSQL := fmt.Sprintf("SELECT COUNT(*) as count from %s where user_id=?", dialect.Quote(query.Target))
+			// removing service accounts from the count
+			if query.Target == dialect.Quote("user") {
```
Comment (kalleep) [inline pkg/services/sqlstore/quota.go:170]: We don't have to make changes here. User quotas only exists on org and global level

File: pkg/services/sqlstore/quota.go [hunk 4]
```diff
@@ -204,6 +218,10 @@ func (ss *SQLStore) GetUserQuotas(ctx context.Context, query *models.GetUserQuot
 			if q.Target != alertRuleTarget || query.UnifiedAlertingEnabled {
 				// get quota used.
 				rawSQL := fmt.Sprintf("SELECT COUNT(*) as count from %s where user_id=?", dialect.Quote(q.Target))
+				// removing service accounts from the count
```
Comment (kalleep) [inline pkg/services/sqlstore/quota.go:221]: Same as above

**Review by kalleep — COMMENTED** [review]
Looks good, but we should remove some changes

File: pkg/services/sqlstore/quota.go (hunk 3 above)
Comment (eleijonmarck) [reply pkg/services/sqlstore/quota.go:170]: I thought though user quota as far as i can tell is calling to get users? 🤔

Comment (eleijonmarck) [reply pkg/services/sqlstore/quota.go:170]: from this point for example? https://github.com/grafana/grafana/blob/main/pkg/services/quota/quotaimpl/quota.go#L117

Comment (papagian) [reply pkg/services/sqlstore/quota.go:170]: agree with @eleijonmarck that we need this: it's used for getting the number of users (on org and global level),

**Review by papagian — DISMISSED** [review]
LGTM

**Review by papagian — APPROVED** [review]
For org quota we [check](https://github.com/grafana/grafana/blob/004bb7689da17806f0263a9503a2599c424a46dc/pkg/services/quota/quotaimpl/quota.go#L147) the `org_user` table instead of the `user` therefore the queries should be like:
`SELECT COUNT(*) AS count FROM org_user WHERE org_id=1 AND user_id NOT IN (SELECT id FROM user WHERE is_service_account = 1);`

**Comment by eleijonmarck** [comment]
@papagian il merge this now. And we can have a look at refactoring this for next week? 👍

**Review by papagian — APPROVED** [review]

**Pull request merged**